src/main/java/org/ml_methods_group/algorithm/sddrar/SDDRARFacade.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 Machine Learning Methods in Software Engineering Group of JetBrains Research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ml_methods_group.algorithm.sddrar;
+
+import org.ml_methods_group.algorithm.sddrar.rules.Rule;
+import org.ml_methods_group.algorithm.sddrar.rules.RuleExtractor;
+import org.ml_methods_group.algorithm.sddrar.rules.RulePack;
+import com.sixrr.metrics.Metric;
+import com.sixrr.metrics.MetricCategory;
+import com.sixrr.metrics.metricModel.MetricsResult;
+import com.sixrr.metrics.metricModel.MetricsRun;
+import com.sixrr.metrics.profile.MetricInstance;
+import com.sixrr.metrics.profile.MetricsProfile;
+import org.apache.commons.math3.linear.MatrixUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+
+public class SDDRARFacade {
+    private static final double MIN_CONFIDENCE = 0.8;
+    private static final double PERCENTAGE_OF_ERROR_THRESHOLD = 0.9;
+
+    public static void trainAndPersistModel(MetricsProfile profile, MetricsRun metricsRun) {
+        DataSet dataSet = extractDataSet(metricsRun);
+        CorrelationFilter.filterByFeatureCorrelationRate(dataSet); //todo пробное удаление
+        List<String> interestingMetrics = dataSet.getFeatureNames();
+        System.out.println(interestingMetrics);
+        SDDRARioHandler.dumpMetrics(interestingMetrics);
+
+        Set<Rule> rules = RuleExtractor.extractRules(dataSet, MIN_CONFIDENCE);
+        RulePack pack = new RulePack(rules, dataSet.getFeatureNames());
+        SDDRARioHandler.dumpRulePack(pack);
+    }
+
+    // приходит уже только с нужными метриками
```

Comment:
This means metricsRun should contains only metrics on which the algorithm was trained. I gues, it should be removed.